$OpenBSD: patch-digikam_utilities_cameragui_gpcamera_cpp,v 1.1 2013/01/19 11:26:02 zhuk Exp $
Fix compilation with GPhoto2. Modelled after this patch for later Digikam: 
http://bugsfiles.kde.org/attachment.cgi?id=73176
--- digikam/utilities/cameragui/gpcamera.cpp.orig	Sat Jan 19 00:26:09 2013
+++ digikam/utilities/cameragui/gpcamera.cpp	Sat Jan 19 00:33:14 2013
@@ -847,6 +847,8 @@ bool GPCamera::uploadItem(const QString& folder, const
 
     errorCode = gp_camera_folder_put_file(d->camera,
                                           QFile::encodeName(folder),
+                                          QFile::encodeName(itemName),
+                                          GP_FILE_TYPE_NORMAL,
                                           cfile,
                                           m_status->context);
     if (errorCode != GP_OK) 
@@ -1108,7 +1110,9 @@ void GPCamera::getSupportedPorts(QStringList& plist)
         for (int i = 0 ; i < numPorts ; i++) 
         {
             gp_port_info_list_get_info( list, i, &info );
-            plist.append( info.path );
+            char *path;
+            gp_port_info_get_name( info, &path );
+            plist.append( path );
         }
     }
 
